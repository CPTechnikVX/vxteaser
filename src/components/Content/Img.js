import React      from 'react';
import PropTypes  from 'prop-types';
import Constants  from '../../utils/Constants';
import classnames from 'classnames';

/**
 * Image component
 */
export default class Img extends React.PureComponent {
	render() {
		const {alt, height, modifier, src, width} = this.props;
		const classList                      = [Constants.ClassName.Image];

		if (modifier) {
			classList.push(modifier);
		}

		return <img className={classnames(classList)} src={src} width={width} height={height} alt={alt} />;
	}
}

Img.propTypes = {
	alt:      PropTypes.string,
	height:   PropTypes.string,
	src:      PropTypes.string,
	modifier: PropTypes.string,
	width:    PropTypes.string,
};
