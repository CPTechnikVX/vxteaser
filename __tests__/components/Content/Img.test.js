import React    from 'react';
import {render} from '@testing-library/react';
import Img      from '../../../src/components/Content/Img';

describe('render picture', () => {
    test('default', () => {
        const {container} = render(
                <Img modifier="-big-pic" />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
