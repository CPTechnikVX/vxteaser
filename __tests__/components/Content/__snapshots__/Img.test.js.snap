// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render picture default 1`] = `
<img
  class="vxteaser-image -big-pic"
/>
`;
